types/serve-static/serve-static-tests.ts

```diff
@@ -26,3 +27,20 @@ serveStatic.mime.define({
     'application/babylonmeshdata': ['babylonmeshdata'],
     'application/fx': ['fx']
 });
+
+serveStatic('/does-not-assume-express', {
+    setHeaders: function(res) {
+        // $ExpectError
+        res.set("foo", "bar");
+    }
+});
+
+http.createServer((req, res) => {
+    serveStatic('/works-with-node-server')(req, res, () => {});
+});
+
+app.use(serveStatic('/infers-express-response-when-passed-to-express-use', {
+    setHeaders: function(res) {
+        res.set('foo', 'bar');
```

Comment:
Please add an `// $ExpectType`  on the next line to verify the contextual type is still `express.Response`.